index.js

```diff
@@ -812,6 +861,32 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
                 }
             }
         });
+        if ($meta.saveAs) {
+            var fileDir = path.dirname($meta.saveAs);
+            return new Promise((resolve, reject) => {
+                fs.mkdir(fileDir, (e) => {
+                    if (!e || e.code === 'EEXIST') {
+                        return resolve();
+                    }
+                    return reject(e);
+                });
+            }).then(function(resolve, reject) {
+                request.stream = true;
+                var ws = fs.createWriteStream($meta.saveAs);
+                transform(request).pipe(ws);
+                request.execute(name);
+                return new Promise(function(resolve, reject) {
+                    ws.on('finish', function() {
+                        return resolve({fileName: $meta.saveAs});
+                    });
+                    ws.on('error', function(err) {
+                        return reject(err);
+                    });
+                });
+            }).catch(function(err) {
```

Comment:
Why did you add this catch? This will result in promise being resolved with the error, which I doubt to be expected.

Reply:
Sorry, Yes. It should be removed and may be handle it on consumer end if needed. Now I have removed the catch section please check it.